### DISCUSSION TITLE
How to make a frame have rounded corners

### DISCUSSION THREAD

**Discussion by Josborne123** [TomSchimansky/CustomTkinter#543] [Q&A]
How do I make this frame have rounded corners? I have tried using corner_radius but that doesn't change anything

![frame](https://user-images.githubusercontent.com/50294428/197188912-f282d466-f7cb-410e-8a4a-e616cb5c33a1.png)

Here is my code for the frame and widgets inside the frame 
```python    
switchframe = customtkinter.CTkFrame(window, width=100, height=40, corner_radius=100)
switchframe.place(relx=0.05, rely=0.17)
switch = customtkinter.CTkSwitch(switchframe, text="", command=mute_switch, text_font=("Comic Sans MS bold", 11), button_color="#6c43cd", fg_color="#b8d2d4", button_hover_color="#a589e8", progress_color="#b8d2d4", height=17)
switch.place(relx=0.23, rely=0.58, anchor="center")
switchtext = customtkinter.CTkLabel(switchframe, text="Audio on", text_font=("Comic Sans MS bold", 10), width=10)
switchtext.place(relx=0.68, rely=0.5, anchor="center")
```

(note: its the actual frame I want to have rounded corners not the elements inside)

**Comment by TomSchimansky** [comment]
Why exactly are you placing the switch at this coordinates: relx=0.23, rely=0.58? The switch background covers the roundedd corners of the frame, you have to give it a padding. I would suggest: switch.pack(padx=8, pady=8) for a corner radius of 8.

**Reply by Josborne123** [reply]
I've decided to get rid of the frame and just place the switch and label in the main window, however I'm still having a similar issue with the label. corner_radius doesn't seem to do anything to the label. 
 
`    switch = customtkinter.CTkSwitch(window, text="", command=mute_switch, text_font=("Comic Sans MS bold", 11), button_color="#6c43cd", fg_color="#b8d2d4", button_hover_color="#a589e8", progress_color="#9fb3b5", height=20, width=40, bg_color="#e88b2e", border_color="#ba91ff")
switch.place(relx=0.13, rely=0.19, anchor="center")
switchtext = customtkinter.CTkLabel(window, text="Audio on", text_font=("Comic Sans MS bold", 10), width=60, bg_color="#6c43cd", corner_radius=10)
switchtext.place(relx=0.13, rely=0.243, anchor="center")`

My updated code (sorry the formatting is weird)

**Reply by TomSchimansky** [reply]
You have to format the code like this:
\```python
code...
\```

And you cannot have rounded corners on the background of a switch, just the switch itself can have rounded corners. You have to use a frame.

**Comment by Akascape** [comment]
@Josborne123 A new solution for this issue: https://github.com/TomSchimansky/CustomTkinter/discussions/2214
